app/controllers/new_threads_controller.rb

```diff
@@ -53,10 +54,12 @@ def create
 
   def edit
     @new_thread = NewThread.friendly.find(params[:id])
+    authorize! :edit, @new_thread
   end
```

Comment:
actually i tried that but without individual authorize statements edit authorization isn't working.i.e. a signed in user is able to edit threads of any other user.also if i use `load_and_authorize_resource` it is causing error due friendly id.